Source/WebCore/Modules/webauthn/fido/Pin.cpp

```diff
@@ -250,16 +323,23 @@ std::optional<TokenRequest> TokenRequest::tryCreate(const CString& pin, const Cr
     ASSERT(!keyPairResult.hasException());
     auto keyPair = keyPairResult.releaseReturnValue();
 
-    // 2. Use ECDH and SHA-256 to compute the shared AES-CBC key.
+    // 2. Use ECDH to compute the shared secret, then apply protocol-specific KDF.
     auto sharedKeyResult = CryptoAlgorithmECDH::platformDeriveBits(downcast<CryptoKeyEC>(*keyPair.privateKey), peerKey);
     if (!sharedKeyResult)
         return std::nullopt;
 
-    auto crypto = PAL::CryptoDigest::create(PAL::CryptoDigest::Algorithm::SHA_256);
-    crypto->addBytes(sharedKeyResult->span());
-    auto sharedKeyHash = crypto->computeHash();
+    auto sharedSecret = deriveProtocolSharedSecret(protocol, *sharedKeyResult);
```

Comment:
It seems like you could `WTFMove(*sharedKeyResult)`, since it's not used again. That would avoid the vector copy inside `deriveProtocolSharedSecret(...)` later.